File: Python/grassseed.py
'''
 *
 *	Problem: "Grass Seed Inc."
 *	Problem ID: "grassseed"
 *	https://open.kattis.com/problems/grassseed
 *
 *	Solution: Python
'''

#-----calcCost-----#
def calcCost():	
	costPerUnit = float(input())
	numLawns = int(input())
	
	totalArea = 0
	for lawn in range(numLawns):
		line = input().split()
		width = float(line[0])
		length = float(line[1])
		
		area = width * length
		totalArea += area
	
	totalCost = totalArea * costPerUnit
	return totalCost

#-----main-----#
output = calcCost()
print(output)

File: Ruby/grassseed.rb
=begin
 *
 *	Problem:		Grass Seed Inc.
 *	Problem ID:	grassseed
 *	url:				https://open.kattis.com/problems/grassseed
 *	Solution:		Ruby
 *
=end

class Solver

	def input_to_float(input)
		input_array = input.chomp.split(" ")
		input_float = input_array.map(&:to_f)
		return input_float
	end

	def solve
		cost_per_unit = gets.chomp.to_f
		number_lawns = gets.chomp.to_i
		
		total_area = 0
		for lawn in 1..number_lawns
			width, length = input_to_float(gets)
			area = width * length
			total_area += area
		end
		
		total_cost = total_area * cost_per_unit
		puts total_cost
	end
	
end

solver = Solver.new
solver.solve
